lib/locale_threads.t

```diff
@@ -54,67 +470,1290 @@ SKIP: { # perl #127708
+                                     . ' utf8::encode($str);'
+                                     . ' no warnings "uninitialized";'
+                                     . ' POSIX::mbtowc(undef, undef);'
+                                     . ' POSIX::mbtowc($value, $str); $value;',
+                           'utf8_only');
+            }
+            if ($Config{'d_wcrtomb'} eq 'define') {
+                add_trials('LC_CTYPE', 'my $value;'
+                                     . ' no warnings "uninitialized";'
+                                     . ' POSIX::wctomb(undef, undef);'
+                                     . ' POSIX::wctomb($value, 0xFF);'
+                                     . ' $value;',
+                           'utf8_only');
+            }
+
+            add_trials('LC_CTYPE',
+                       'no warnings "locale";'
+                     . ' my $uc = CORE::uc join "", map { chr } (0..255);'
+                     . ' my $fc = quotemeta CORE::fc $uc;'
+                     . ' $uc =~ / \A $fc \z /xi;');
+            next;
+        }
+
+        if ($category eq 'LC_MESSAGES') {
+            add_trials('LC_MESSAGES',
+                     "join \"\n\", map { \$! = \$_; \"\$!\" } ($msg_catalog)");
+            add_trials('LC_MESSAGES',
+                  'use I18N::Langinfo qw(langinfo YESSTR NOSTR YESEXPR NOEXPR);'
+                . ' no warnings "uninitialized";'
+                . ' join ",",'
+                . '     map { langinfo($_) } YESSTR, NOSTR, YESEXPR, NOEXPR;');
+            next;
+        }
+
+        if ($category eq 'LC_MONETARY') {
+            if ($has_localeconv) {
+                add_trials('LC_MONETARY', "localeconv()->{currency_symbol}");
+            }
+            add_trials('LC_MONETARY',
+                       'use I18N::Langinfo qw(langinfo CRNCYSTR);'
+                    . ' no warnings "uninitialized";'
+                    . ' join "|",  map { langinfo($_) } CRNCYSTR;');
+            next;
+        }
+
+        if ($category eq 'LC_NUMERIC') {
+            if ($has_localeconv) {
+                add_trials('LC_NUMERIC', "no warnings; 'uninitialised';"
+                                       . " join '|',"
+                                       . " localeconv()->{decimal_point},"
+                                       . " localeconv()->{thousands_sep}");
+            }
+            add_trials('LC_NUMERIC',
+                       'use I18N::Langinfo qw(langinfo RADIXCHAR THOUSEP);'
+                     . ' no warnings "uninitialized";'
+                     . ' join "|",  map { langinfo($_) } RADIXCHAR, THOUSEP;');
+
+            # Use a variable to avoid runtime bugs being hidden by constant
+            # folding
+            add_trials('LC_NUMERIC', 'my $in = 4.2; sprintf("%g", $in)');
+            next;
+        }
+
+        if ($category eq 'LC_TIME') {
+            add_trials('LC_TIME', "POSIX::strftime($strftime_args)");
+            add_trials('LC_TIME',
```

Comment:
This might be more readable as a heredoc.

Reply:
I don't understand how you mean this to be

Reply:
Presumably something like:
```perl
add_trials('LC_TIME', <<'END_OF_CODE');
use I18N::Langinfo qw(langinfo
    ABDAY_1 ABDAY_2 ABDAY_3 ABDAY_4 ABDAY_5 ABDAY_6 ABDAY_7
    ABMON_1 ABMON_2 ABMON_3 ABMON_4 ABMON_5 ABMON_6
    ABMON_7 ABMON_8 ABMON_9 ABMON_10 ABMON_11 ABMON_12
    DAY_1 DAY_2 DAY_3 DAY_4 DAY_5 DAY_6 DAY_7
    MON_1 MON_2 MON_3 MON_4 MON_5 MON_6
    MON_7 MON_8 MON_9 MON_10 MON_11 MON_12
    D_FMT D_T_FMT T_FMT);
no warnings "uninitialized";
join "|",
     map { langinfo($_) }
           ABDAY_1,ABDAY_2,ABDAY_3,ABDAY_4,ABDAY_5,
           ABDAY_6,ABDAY_7,
           ABMON_1,ABMON_2,ABMON_3,ABMON_4,ABMON_5,
           ABMON_6, ABMON_7,ABMON_8,ABMON_9,ABMON_10,
           ABMON_11,ABMON_12,
           DAY_1,DAY_2,DAY_3,DAY_4,DAY_5,DAY_6,DAY_7,
           MON_1,MON_2,MON_3,MON_4,MON_5,MON_6, MON_7,
           MON_8,MON_9,MON_10,MON_11,MON_12,
           D_FMT,D_T_FMT,T_FMT;
END_OF_CODE
```

Reply:
thanks; changed to @mauke suggestion